iocore/eventsystem/I_Continuation.h

```diff
@@ -63,6 +63,26 @@ extern EThread *this_event_thread();
 
 typedef int (Continuation::*ContinuationHandler)(int event, void *data);
 
+// Convert event handler pointer fp to type ContinuationHandler, but with a compiler error if class C is not
+// derived from the class Continuation.
```

Comment:
Can we check this at compile time & avoid casting twice like this?
```
template <class C, typename T>
constexpr ContinuationHandler
continuation_handler_void_ptr(int (C::*fp)(int, T *))
{
  static_assert(std::is_base_of<Continuation, C> && std::is_convertible<C, Continuation>);
  return reinterpret_cast<fp>;
}
```

BTW, we can check this clearly with the c++20 concept
```
template <class C, typename T>
requires std::derived_from(C, Continuation)
constexpr ContinuationHandler
continuation_handler_void_ptr(int (C::*fp)(int, T *))
{
```

Reply:
It is a constexpr function, it will be evaluated at compile time, and cause no additional instructions in the executable.

Reply:
Also, your alternative does not check that the handler function has the right parameters (an int and a pointer).

Reply:
Ah, I missed the `constexpr` part. Great :+1: